design/FY2021/NFP-Spaces.md

```diff
@@ -0,0 +1,234 @@
+        Space 2
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+    
+
+### Definitions
+
+ * Surface - A geometric plane which is attached to a Space. 
+   * A Surface can be opaque, transparent, or an air boundary.
+   * Each Surface belongs to one Space.
+
+ * Space - A collection of one or more Surfaces and internal gains.
+   * Each Space belongs to one Zone (explicitly user-assigned).
+   * The Spaces in a Zone are lumped together for the Zone heat balance.
+   * There is no heat balance at the Space level.
+   * Each Space belongs to one Enclosure (implicitly assigned).
+
+ * Zone - An air mass connecting Surfaces, internal gains, and HVAC equipment for heat balance and HVAC control.
+   * Each Zone is comprised of one or more Spaces.
+   * All Surfaces and internal gains associated with the Spaces are included in the Zone.
+   * The Zone heat balance does not change: it has an air node (or a Room Air Model) and includes all Surfaces and 
+   internal gains from its Spaces plus any HVAC which is attached to the Zone.
+
+ * Enclosure - A continuous volume connecting Surfaces for radiant, solar, and daylighting exchange.
+   * Each Enclosure is comprised of one or more Spaces.
+   * There is one Enclosure for each Space unless there are air boundary surfaces 
+   which group multiple Spaces into a single Enclosure.
+   * All Surfaces and internal radiant gains associated with the Spaces are included in the Enclosure.
+   * Enclosures only distribute radiant (thermal), solar, and visible energy to and from the Surfaces.
+   * There is no full heat balance at the Enclosure level. Each Enclosure only balances the radiant/solar flux on each 
+   Surface. These fluxes then become part of the Surface inside heat balance.
+
+## Proposed Input Approach ##
+Very minimal changes to current inputs.
+
+ * Old Zone object becomes Space object plus some tags.
+ * New Zone object has a name and a list of Spaces.
+ * Surfaces will reference a Space instead of a Zone.
+ * ZoneHVAC and Thermostats remain as-is and continue to reference a Zone or ZoneList.
+ * New SpaceList object (equivalent to ZoneList, but for Spaces).
+ * Internal gains reference Space or Spacelist (instead of Zone or ZoneList).
+ * ZoneInfiltration and ZoneVentilation remain at the zone level.
+
+### Proposed objects:
+
+    Space,  !- Same fields as old Zone object plus new tags at the end
+      Name,
+      Origin,
+      Multiplier,
+      Ceiling Height,
+      Volume,
+      Floor Area,
+      Convection algorithms,
+      Part of Total Floor Area,
+      User-defined Tags for Space Types (for reporting purposes only at this point)
+    
+    Lighting/People/Equipment
+      Name,
+      Space or SpaceList Name, !- Replace current Zone or ZoneList Name
+      Schedule Name,
+      ...
+    
+    ZoneVentilation and ZoneInfiltration
+      Name,
+      Zone or ZoneList Name, (or should these more to the Space level and be renamed?)
```

Comment:
I haven't thought about this before. I have in the past included a part of a hallway with the adjacent office space as a single zone, just to save time on data entry. The hallway in a commercial building would have a different infiltration rate than other spaces since it would be connected to external doors, and would typically not have a thermostat. Of course you could add a thermostat in a simulation but that would defeat the purpose of "zoning" spaces around a common thermostat. So if this is left as a zone object the user would need to correctly group spaces of like ventilation/infiltration to a single zone instead of grouping spaces near a thermostat to a single zone. So it seems to me ventilation/infiltration should be applied at the space level.